secure_time/storage/secure_time_nvstore.cpp

```diff
@@ -0,0 +1,125 @@
+/* Copyright (c) 2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <secure_time_utils.h>
+#include "secure_time_storage.h"
+#include "secure_time_client_spe.h"
+#include "secure_time_impl.h"
+#include "nvstore.h"
+#include "mbed_toolchain.h"
+#include "mbed_error.h"
+
+#if !NVSTORE_ENABLED
+#error [NOT_SUPPORTED] NVSTORE needs to be enabled
```

Comment:
I think this is causing build to fail. This approach is fine for tests but not for a regular file. It should build otherwise it is causing a failure for any target that does not support nvstore. Rather this would be protected only

Reply:
We have similar approach in nvstore, its protected with `NVSTORE_ENABLED`